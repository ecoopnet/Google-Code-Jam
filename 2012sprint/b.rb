#!/usr/bin/env ruby -Ku
# Usage: cat problems | program.rb

class Solver
	def solve line
		items = line.split(" ")
		@K = items.shift.to_i
		@S = items.shift
		i = 0
		results = []
		while i <= @S.length - @K
			v = @S[i,@K]
			if (@S.rindex v) > i
				results.push v 
			end
			i+=1
		end
		if results.length == 0
			return "NONE"
		end
		return results.sort.uniq.join " "
	end
end

solver = Solver.new
number = ARGF.gets
i = 0
while line = ARGF.gets
	line.strip!
	if line == ""
		next
	end
	i+=1
	print "Case #",i,": ", solver.solve(line),"\n"
end
