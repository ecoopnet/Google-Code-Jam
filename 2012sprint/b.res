Case #1: NONE
Case #2: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #3: 11241 11323 11412 11453 11555 12142 12144 12222 12314 12513 13213 13525 13552 14154 14255 14412 14413 14535 15211 15234 15414 15444 15524 15534 21115 21211 21251 21415 21421 21441 21511 22144 22151 22233 22335 22531 23141 23143 23352 23433 24255 24335 24453 24542 25333 25342 25511 25554 31352 31355 31414 32114 32343 32425 32555 33152 33221 33234 33242 33332 33521 34135 34154 34314 34334 35251 35253 35334 35441 35521 35553 41154 41214 41311 41323 41354 41451 41553 42225 42341 42343 42433 42552 42553 43135 43314 43334 43341 43533 44131 44132 44531 45244 45255 45314 45325 45355 45445 51124 51141 51253 51441 51521 51535 52111 52125 52214 52445 53255 53341 53415 53555 54121 54333 54442 55152 55225 55232 55243 55335 55341 55512 55533 55544 55545
Case #4: 1111 1112 1113 1114 1121 1122 1124 1131 1132 1133 1134 1141 1142 1143 1144 1211 1212 1213 1214 1221 1222 1223 1224 1231 1234 1241 1242 1243 1244 1311 1312 1313 1314 1322 1323 1324 1331 1332 1333 1341 1342 1343 1344 1411 1412 1413 1414 1421 1422 1423 1424 1431 1432 1434 1441 1442 1443 1444 2111 2112 2113 2114 2122 2124 2131 2132 2133 2134 2141 2142 2143 2144 2211 2212 2214 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2312 2313 2314 2321 2322 2324 2331 2332 2333 2341 2342 2343 2411 2412 2413 2414 2421 2422 2423 2424 2431 2432 2433 2434 2441 2442 2443 2444 3111 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3211 3212 3213 3214 3222 3224 3231 3232 3233 3234 3241 3242 3243 3244 3311 3312 3313 3314 3321 3324 3331 3332 3333 3334 3341 3342 3344 3412 3413 3414 3421 3422 3423 3424 3432 3433 3434 3441 3442 3444 4111 4112 4113 4114 4121 4122 4123 4124 4131 4132 4133 4134 4141 4142 4143 4144 4211 4212 4213 4214 4221 4222 4223 4224 4232 4233 4234 4241 4242 4243 4244 4311 4312 4313 4321 4322 4323 4324 4331 4332 4333 4334 4341 4342 4343 4344 4411 4413 4414 4421 4422 4423 4424 4432 4433 4434 4441 4442 4443 4444
Case #5: 1
Case #6: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #7: 1
Case #8: NONE
Case #9: 11 12 21 22
Case #10: NONE
Case #11: NONE
Case #12: NONE
Case #13: NONE
Case #14: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #15: 11211 11222 11532 11554 12113 12132 12411 12445 12455 12541 12544 13221 13235 13311 13343 14453 14533 15125 15223 15224 15251 15322 15545 21153 21323 21512 21521 21522 22152 22234 22334 22342 22423 23132 23144 23152 23242 23334 23343 23421 23431 23542 24112 24231 24235 24254 25132 25215 25222 25244 25333 25414 25452 25522 31115 31143 31211 31254 31423 31445 31525 32143 32212 32313 32324 32354 33112 33212 33343 33422 33431 33433 33512 34312 34314 34332 34412 34512 35252 35334 35343 35424 35515 41213 41323 41522 42331 42335 42344 42425 42512 42533 42545 43111 43114 43125 43321 44311 44312 44454 44533 44535 45242 45332 45333 45335 45355 45425 51213 51241 51255 51512 52152 52222 52223 52235 52423 52444 52521 53332 53351 54121 54143 54233 54242 54431 54445 54542 55124 55151 55222 55453 55454 55512
Case #16: 3636441
Case #17: 11 12 21 22
Case #18: 11 12 21 22
Case #19: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #20: 11132 11233 11251 11321 11432 12122 12213 12221 12243 12415 12514 13124 13125 13334 13412 13554 14153 14244 14313 14315 14412 15325 15345 15412 15431 15433 21125 21135 21222 21252 21341 21415 21443 21543 21555 22225 22254 22445 22545 23141 23321 23545 24311 24353 24411 24433 24512 24533 25422 25453 31143 31212 31253 31331 31354 31532 32114 32324 32332 32335 32431 33114 33213 33342 33425 33441 33515 33545 34251 35124 35215 35331 35424 35452 35455 35545 41122 41224 41312 41534 42311 42323 42425 42443 42515 43143 43153 43232 43342 43344 43453 44112 44125 44235 44334 44335 44541 45133 45142 45315 45352 51241 51421 51532 52435 52441 53214 53311 53315 53354 53451 53521 54112 54131 54231 54345 54511 54535 55245 55335 55442 55533
Case #21: NONE
Case #22: NONE
Case #23: 153124 154426 162415 262141 322621 324624 331511 345464 415455 433151 454225 454646 531243 621413 624154 636232 642153
Case #24: NONE
Case #25: 1474344
Case #26: NONE
Case #27: 11131 11253 11312 11315 11353 11413 12253 12422 12515 13225 13241 13315 13331 13354 14144 14253 14331 14425 14554 15211 15451 22245 22252 22313 22431 22453 22524 22532 22533 23133 23354 23453 23554 24252 24313 24341 24445 25152 25214 25234 25241 25245 25333 25443 25552 31224 31251 31254 31333 31411 32224 32243 32345 32351 33125 33133 33154 33235 33323 33541 33543 34314 34451 34512 34533 35145 35222 35413 35434 35455 41135 41242 41331 41425 41442 42325 42534 43125 43133 43224 43233 43313 43411 43425 44322 44511 45114 45124 45142 45314 45323 45543 51141 51424 51425 51455 51523 52115 52143 53312 53355 53431 54124 54132 54233 55121 55154 55255
Case #28: 122133 123351 125526 232364 234445 412324 551665 631136 631463 633445
Case #29: 11 12 21 22
Case #30: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #31: NONE
Case #32: 11 12 21 22
Case #33: 1111 1112 1113 1114 1121 1122 1123 1131 1132 1133 1134 1141 1142 1143 1211 1212 1213 1214 1221 1222 1223 1224 1232 1233 1234 1241 1242 1243 1244 1311 1313 1314 1321 1322 1323 1324 1331 1333 1334 1341 1342 1343 1344 1411 1412 1413 1414 1422 1423 1424 1431 1432 1433 1434 1442 1443 2111 2113 2114 2121 2122 2123 2124 2131 2132 2133 2134 2141 2143 2144 2211 2213 2223 2224 2231 2232 2234 2241 2242 2243 2244 2311 2312 2313 2314 2321 2322 2323 2324 2331 2332 2333 2334 2341 2342 2343 2344 2411 2412 2413 2414 2421 2422 2423 2424 2431 2432 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3143 3144 3211 3212 3213 3214 3221 3223 3224 3231 3232 3233 3234 3241 3242 3243 3244 3311 3312 3313 3314 3321 3322 3324 3331 3332 3333 3334 3341 3342 3343 3344 3411 3412 3413 3414 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4114 4121 4122 4123 4124 4131 4132 4133 4141 4142 4143 4144 4211 4212 4213 4223 4224 4231 4232 4233 4234 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4323 4324 4331 4332 4333 4334 4341 4342 4343 4344 4411 4413 4421 4422 4423 4424 4431 4432 4434 4442 4443 4444
Case #34: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #35: 11 12 21 22
Case #36: 1111 1112 1113 1114 1121 1122 1123 1124 1131 1132 1133 1134 1141 1142 1143 1144 1211 1214 1221 1222 1223 1224 1231 1232 1233 1234 1241 1242 1243 1311 1312 1313 1321 1322 1323 1324 1331 1332 1333 1334 1341 1342 1343 1344 1411 1413 1414 1421 1422 1423 1424 1431 1432 1433 1434 1441 1443 1444 2111 2112 2113 2114 2121 2122 2123 2131 2132 2133 2134 2141 2142 2143 2144 2211 2213 2214 2221 2222 2223 2224 2231 2233 2241 2242 2243 2244 2311 2313 2314 2321 2323 2324 2331 2332 2334 2341 2342 2343 2344 2411 2412 2413 2414 2421 2422 2424 2431 2432 2433 2434 2441 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3143 3144 3211 3212 3213 3214 3221 3222 3224 3231 3232 3233 3241 3242 3243 3244 3311 3312 3313 3314 3322 3323 3324 3331 3333 3334 3341 3343 3344 3411 3412 3413 3414 3421 3422 3424 3431 3432 3433 3441 3442 3443 3444 4111 4113 4114 4122 4123 4131 4132 4134 4141 4142 4143 4144 4211 4212 4213 4214 4221 4222 4223 4232 4233 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4323 4324 4331 4332 4333 4334 4341 4342 4343 4411 4412 4413 4414 4421 4422 4423 4431 4432 4433 4434 4441 4442 4443 4444
Case #37: NONE
Case #38: 1111 1112 1113 1121 1122 1123 1124 1131 1132 1133 1134 1142 1211 1212 1213 1214 1221 1222 1223 1224 1231 1232 1233 1234 1241 1242 1244 1311 1312 1313 1314 1322 1323 1324 1332 1333 1334 1341 1342 1343 1344 1411 1412 1413 1414 1421 1422 1423 1431 1432 1433 1441 1443 1444 2112 2113 2114 2121 2122 2123 2124 2131 2132 2133 2134 2141 2142 2143 2211 2212 2213 2214 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2311 2313 2314 2321 2322 2323 2324 2331 2332 2333 2334 2341 2342 2343 2344 2411 2412 2413 2414 2421 2422 2424 2431 2432 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3143 3144 3211 3212 3213 3214 3222 3223 3224 3231 3232 3233 3234 3242 3243 3244 3311 3312 3313 3314 3321 3322 3323 3324 3331 3332 3333 3334 3341 3342 3344 3411 3413 3421 3422 3423 3424 3431 3433 3434 3441 3442 3444 4111 4112 4121 4122 4123 4131 4132 4133 4134 4141 4142 4144 4211 4212 4213 4214 4221 4222 4223 4224 4231 4232 4233 4234 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4324 4331 4333 4334 4342 4343 4411 4412 4413 4414 4421 4422 4423 4424 4431 4432 4433 4434 4441 4442 4443 4444
Case #39: 2227277 2272772
Case #40: 7423724
Case #41: 11131 11142 11313 11422 11452 11533 12122 12211 12333 12435 13111 13131 13133 13141 13211 13233 13435 14113 14214 14354 14414 14424 14432 14525 21222 21243 21311 21433 22231 22341 22441 22443 22513 22535 22553 23224 23332 23414 23422 24114 24123 24212 24324 24352 24411 25115 25121 25134 25253 25354 25444 25534 31113 31114 31131 31233 31313 31314 31441 31545 32132 32244 32323 32331 33213 33245 33354 33542 34115 34212 34351 34354 35323 35413 35455 41132 41144 41145 41152 41153 41233 41411 41514 42122 42124 42244 42534 43215 43231 43241 43251 43541 43544 44115 44144 44241 44324 44325 44452 44525 45225 45253 45522 51142 51153 51555 52223 52234 52251 52255 52412 52533 52535 52544 53131 53411 53522 54113 54412 54552 55222 55223 55341 55445 55552
Case #42: 1112 1113 1114 1121 1122 1123 1124 1131 1132 1133 1134 1141 1142 1143 1211 1212 1213 1214 1221 1222 1223 1224 1231 1232 1233 1234 1241 1243 1244 1311 1312 1313 1314 1321 1322 1331 1333 1334 1341 1342 1343 1344 1411 1412 1413 1414 1421 1423 1424 1431 1432 1433 1434 1441 1442 1443 2111 2112 2113 2114 2121 2122 2124 2131 2132 2133 2134 2141 2142 2144 2211 2212 2214 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2311 2312 2313 2314 2321 2323 2324 2331 2332 2333 2334 2341 2342 2344 2411 2413 2414 2422 2423 2424 2431 2432 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3143 3211 3212 3213 3214 3221 3223 3224 3231 3232 3233 3234 3241 3242 3243 3244 3311 3312 3313 3314 3321 3323 3324 3331 3332 3333 3341 3342 3343 3344 3411 3412 3413 3421 3422 3423 3424 3431 3432 3433 3434 3441 3443 3444 4111 4112 4113 4114 4121 4122 4123 4131 4132 4134 4141 4144 4211 4212 4213 4214 4221 4222 4223 4224 4231 4232 4234 4241 4242 4243 4244 4311 4312 4313 4321 4322 4323 4324 4331 4332 4333 4334 4342 4343 4344 4412 4413 4414 4421 4422 4424 4431 4432 4433 4434 4441 4442 4443 4444
Case #43: 211616 236134 335646 353335 412333 566456 665614
Case #44: 146244 226141 252236 341462 414624 551545 643442 646512
Case #45: 1
Case #46: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #47: 355615 654213
Case #48: 1
Case #49: 1
Case #50: 126114 156534 164463 256651 336436 366416 451366 516332 522146 663623 663642
Case #51: NONE
Case #52: NONE
Case #53: 11245 11311 11543 12245 12331 12354 12415 12424 12452 13123 13255 13423 13544 14141 14522 15221 15225 15323 15422 15431 21124 21223 21414 22231 22312 22514 23123 23124 23135 23533 23542 24242 24252 24525 24535 25145 25343 25512 25523 25544 31154 31213 31233 31235 31354 31535 32231 32312 32322 32551 32552 32554 33231 33255 33323 33324 33335 33355 34354 35152 35333 35424 35444 35523 41154 41452 41511 42223 42422 42521 43122 43145 43435 43544 44145 44252 44343 44354 44435 44544 44551 45212 45445 45453 45511 45514 45522 45531 51145 51151 51224 51522 51543 52112 52215 52243 52251 52253 52545 53232 53255 54441 54443 54454 54523 54532 55143 55235 55442 55454
Case #54: 11332 11341 12233 12325 12442 13411 14152 14423 14523 15153 15221 15415 15451 15553 21145 21222 21424 21453 21552 22122 22142 22151 22314 22332 22444 22513 24423 24432 24445 24551 25131 25133 25155 31125 31142 32122 32145 32321 32344 32351 32445 32533 33235 33354 33543 34113 34114 34154 34211 34251 34311 34332 35154 35242 35341 41134 41143 41442 41515 42155 42215 42441 42455 42515 43135 43143 43244 43421 43431 44232 44245 44321 44351 44521 44542 44554 45212 45231 45233 45323 45441 45511 45522 45544 51154 51232 51312 51554 52214 52244 52444 53231 54234 54325 54414 54523 55244 55445
Case #55: 1
Case #56: NONE
Case #57: 11243 11522 12353 12355 13113 13153 13345 13422 13455 13521 13542 14251 14314 14553 15314 15423 15441 15541 21235 21342 21431 21554 22224 22344 22431 23235 23445 23515 24213 24255 24313 24324 24355 24451 25112 25345 25434 25532 31131 31152 31243 31334 31421 31434 31455 32155 32323 32343 32351 32553 33115 33452 34131 34225 34242 34255 34322 34333 34452 34523 34552 35151 35212 41315 41441 41543 42123 42134 42253 42323 42425 42454 42511 42543 42551 42553 43124 43221 43234 43242 43311 43413 43543 43545 44354 44423 44512 45134 45213 45343 45555 51134 51215 51514 51544 52112 52322 52454 53145 53215 53433 53455 54123 54154 54232 54345 54354 54435 55121 55245 55321 55342 55412 55512
Case #58: 11 12 21 22
Case #59: NONE
Case #60: 11 12 21 22
Case #61: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #62: 135263 146656 163142 455146 464244 563135 614421 642156 666643
Case #63: 7554257
Case #64: 1111 1112 1113 1114 1121 1122 1123 1124 1132 1134 1141 1142 1143 1144 1211 1212 1213 1214 1221 1222 1223 1224 1231 1232 1233 1234 1241 1242 1243 1244 1313 1314 1321 1322 1323 1324 1331 1332 1333 1341 1342 1343 1344 1411 1412 1413 1414 1421 1422 1423 1424 1431 1432 1433 1434 1442 1443 1444 2111 2112 2114 2121 2122 2123 2124 2132 2133 2134 2141 2142 2143 2144 2211 2212 2213 2214 2221 2222 2223 2224 2231 2232 2233 2241 2242 2244 2311 2312 2313 2314 2321 2322 2323 2324 2331 2332 2334 2341 2344 2412 2413 2414 2421 2422 2424 2431 2432 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3144 3211 3212 3213 3214 3221 3222 3223 3224 3231 3232 3233 3241 3242 3243 3244 3311 3313 3314 3321 3322 3323 3324 3331 3341 3342 3343 3344 3411 3412 3414 3421 3422 3424 3431 3432 3433 3434 3441 3442 3443 3444 4112 4113 4114 4121 4122 4123 4124 4131 4133 4134 4141 4142 4143 4144 4211 4212 4213 4221 4222 4223 4224 4231 4233 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4323 4324 4331 4332 4334 4341 4342 4343 4344 4411 4412 4413 4414 4421 4422 4423 4424 4431 4433 4434 4441 4442 4443 4444
Case #65: 234245 266616 323444 333243 336344 345226 466116 611421 644314
Case #66: 11125 11154 11253 11444 12135 12155 12355 12433 12515 13152 13211 13321 13423 13513 13545 14132 14141 14144 14441 15141 15151 15153 15233 15322 15341 15443 15511 15515 15521 15524 15531 15554 21112 21143 21155 21332 21333 21515 21532 21551 22123 22153 22345 23213 23325 23422 23441 23454 23531 25151 25152 25155 31133 31342 32111 32151 32232 32234 32342 32532 33214 33223 33253 34123 34225 34231 34233 34334 34414 34425 35321 35451 41414 41431 41551 41553 42133 42251 42455 42515 42533 43251 43342 43442 44143 44155 44415 45121 45153 45515 45521 51213 51322 51414 51512 51514 51515 51523 51534 51552 51555 52154 52332 52342 52424 52442 53134 53211 53221 53322 54344 54444 54515 55111 55151 55211 55244 55353 55434 55544
Case #67: 1111 1112 1113 1114 1121 1122 1123 1124 1132 1133 1134 1141 1142 1143 1144 1211 1212 1213 1214 1221 1222 1223 1224 1233 1234 1241 1242 1243 1244 1311 1312 1313 1314 1321 1323 1324 1331 1332 1333 1334 1342 1343 1344 1411 1412 1413 1414 1421 1422 1423 1424 1431 1432 1441 1442 1443 1444 2111 2112 2113 2121 2122 2124 2131 2132 2133 2134 2141 2142 2144 2211 2212 2213 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2311 2312 2313 2314 2322 2323 2324 2331 2332 2333 2334 2341 2343 2411 2412 2413 2414 2421 2422 2423 2424 2431 2432 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3134 3141 3142 3143 3144 3212 3213 3214 3221 3222 3223 3224 3231 3232 3233 3234 3243 3244 3311 3312 3313 3314 3322 3323 3324 3331 3332 3334 3341 3342 3343 3344 3411 3413 3414 3421 3422 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4113 4114 4121 4122 4123 4124 4131 4132 4133 4134 4141 4142 4143 4144 4211 4212 4213 4214 4221 4222 4223 4224 4231 4232 4233 4234 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4323 4324 4331 4334 4341 4342 4344 4411 4412 4413 4414 4421 4422 4423 4424 4431 4432 4433 4434 4442 4443 4444
Case #68: NONE
Case #69: NONE
Case #70: NONE
Case #71: 136355 142321 144141 321356 332162 425334 442424 464213 532146
Case #72: NONE
Case #73: 1
Case #74: 1111 1112 1113 1114 1121 1122 1124 1132 1133 1134 1141 1142 1143 1144 1211 1212 1213 1214 1221 1222 1223 1224 1231 1232 1234 1241 1242 1244 1311 1312 1313 1314 1321 1322 1331 1332 1333 1334 1341 1342 1343 1344 1411 1412 1414 1421 1422 1423 1424 1431 1432 1433 1434 1441 1442 1444 2111 2112 2113 2114 2121 2122 2123 2124 2131 2133 2134 2141 2142 2143 2144 2212 2213 2214 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2311 2312 2313 2321 2322 2323 2324 2331 2332 2333 2334 2342 2343 2344 2411 2412 2413 2414 2421 2422 2423 2424 2431 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3143 3144 3211 3212 3214 3221 3223 3224 3232 3233 3234 3241 3244 3311 3312 3313 3314 3321 3322 3323 3324 3331 3332 3333 3334 3342 3343 3344 3411 3413 3414 3421 3422 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4113 4114 4121 4122 4123 4131 4133 4134 4141 4142 4143 4144 4211 4212 4213 4214 4221 4222 4223 4224 4231 4232 4233 4234 4242 4243 4244 4311 4312 4313 4314 4321 4322 4324 4331 4333 4341 4342 4343 4344 4411 4412 4413 4414 4421 4422 4423 4424 4431 4432 4433 4434 4441 4442 4443 4444
Case #75: NONE
Case #76: 122212 162514 211546 241666 416664 445255 546166 616146
Case #77: 1111 1112 1113 1114 1121 1123 1124 1131 1132 1133 1134 1142 1143 1144 1212 1213 1214 1221 1224 1231 1233 1234 1241 1242 1243 1244 1311 1312 1313 1321 1322 1323 1324 1331 1333 1334 1342 1343 1344 1411 1412 1413 1414 1421 1422 1423 1424 1431 1432 1433 1434 1441 1442 1443 2111 2112 2113 2121 2122 2123 2124 2131 2132 2133 2134 2141 2142 2143 2144 2212 2213 2214 2221 2222 2223 2233 2234 2241 2242 2244 2311 2313 2314 2322 2323 2324 2331 2332 2333 2334 2342 2343 2344 2412 2413 2414 2422 2423 2424 2431 2432 2433 2434 2441 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3211 3212 3213 3222 3223 3231 3232 3233 3234 3241 3242 3243 3244 3311 3312 3313 3314 3321 3322 3324 3331 3332 3334 3341 3342 3343 3344 3411 3412 3414 3421 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4113 4114 4122 4123 4124 4131 4132 4133 4134 4141 4142 4143 4144 4212 4213 4214 4221 4222 4231 4232 4234 4241 4242 4243 4244 4311 4312 4313 4314 4322 4323 4324 4331 4332 4333 4334 4341 4342 4343 4344 4411 4412 4413 4414 4421 4422 4423 4424 4432 4433 4434 4441 4442 4443 4444
Case #78: NONE
Case #79: 1
Case #80: 11 12 21 22
Case #81: 11 12 21 22
Case #82: 235426 236363 245425 354421 361555 363126 431143 442133 446426 536155 544213 544614 562446 615536 631261 666666
Case #83: 1
Case #84: 11211 11343 11421 11455 11551 12114 12553 13141 13252 13443 13454 13531 14215 14343 14344 15212 15345 15423 21124 21235 21315 21423 23412 23442 23511 23534 23545 24142 24224 24341 24352 24554 25252 25411 25432 31255 31411 31434 31451 31534 32332 32351 32422 32525 33132 33215 33242 33313 33331 34121 34235 34241 34254 34325 34333 34432 34441 35115 35315 35325 35453 35515 41135 41325 41434 42131 42314 42323 42353 42354 42432 42434 42541 43143 43233 43244 43252 43411 43424 43433 44143 44334 44414 45135 45212 45443 45525 51142 51345 51353 52123 52125 52444 52455 52524 52543 53125 53143 53153 53255 53352 53425 54143 54234 54243 54431 55114 55153 55154 55423 55542
Case #85: 1111 1112 1113 1121 1122 1123 1124 1131 1132 1133 1134 1141 1143 1144 1211 1212 1213 1214 1221 1222 1223 1224 1231 1232 1233 1234 1241 1242 1243 1244 1311 1312 1313 1314 1321 1322 1323 1324 1331 1333 1341 1342 1343 1344 1412 1414 1421 1422 1424 1431 1432 1434 1441 1442 1444 2111 2112 2113 2121 2122 2123 2124 2131 2132 2133 2134 2142 2143 2144 2211 2212 2213 2214 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2311 2313 2314 2321 2322 2324 2331 2332 2333 2334 2342 2343 2344 2411 2412 2413 2414 2422 2423 2431 2432 2433 2434 2442 2443 3111 3112 3113 3114 3121 3122 3123 3124 3132 3134 3141 3142 3143 3144 3211 3212 3213 3214 3221 3222 3223 3224 3232 3233 3234 3241 3242 3243 3244 3311 3312 3313 3314 3321 3322 3323 3324 3331 3333 3334 3341 3342 3344 3411 3412 3413 3414 3421 3422 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4113 4114 4121 4122 4123 4124 4131 4132 4133 4134 4142 4143 4144 4211 4212 4213 4222 4223 4224 4231 4232 4233 4234 4243 4244 4311 4312 4313 4314 4322 4323 4332 4333 4334 4341 4342 4343 4344 4411 4412 4413 4414 4421 4422 4423 4424 4431 4432 4433 4441 4442 4443 4444
Case #86: 1
Case #87: NONE
Case #88: 1111 1112 1113 1114 1121 1122 1123 1124 1131 1133 1134 1141 1142 1143 1144 1211 1213 1221 1222 1223 1224 1231 1232 1233 1234 1241 1242 1243 1244 1311 1312 1313 1314 1321 1322 1323 1331 1332 1333 1334 1341 1342 1343 1344 1411 1412 1413 1414 1421 1422 1424 1431 1432 1433 1434 1441 1442 1443 1444 2111 2112 2113 2114 2123 2124 2131 2132 2134 2141 2142 2143 2144 2211 2212 2213 2214 2221 2222 2223 2224 2231 2232 2233 2234 2241 2242 2243 2244 2313 2314 2321 2322 2323 2324 2331 2332 2333 2334 2342 2343 2344 2411 2412 2413 2414 2421 2422 2423 2424 2431 2433 2434 2441 2442 2443 2444 3111 3112 3113 3114 3121 3122 3123 3124 3131 3132 3133 3134 3141 3142 3143 3144 3211 3212 3213 3214 3221 3222 3223 3224 3231 3232 3233 3234 3241 3243 3244 3311 3312 3313 3314 3322 3323 3324 3331 3332 3334 3341 3342 3343 3344 3411 3412 3414 3421 3422 3423 3424 3431 3432 3433 3434 3441 3442 3443 3444 4111 4112 4113 4121 4122 4124 4131 4132 4133 4134 4141 4142 4143 4144 4211 4212 4213 4214 4221 4222 4223 4224 4231 4232 4233 4234 4241 4242 4243 4244 4311 4312 4313 4314 4321 4322 4323 4324 4331 4333 4334 4341 4342 4343 4344 4411 4412 4413 4414 4421 4422 4423 4424 4431 4432 4433 4434 4441 4442 4443 4444
Case #89: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #90: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
Case #91: NONE
Case #92: 11111 11331 12111 12113 12233 12243 12431 12554 13222 13223 13313 13314 14124 14241 14414 14534 15133 15334 15335 15343 15524 21211 21442 22324 22332 22333 22535 23132 23243 23324 23343 23425 24142 24314 24323 24355 24512 25234 25355 25451 25544 31211 31255 31322 31412 31522 32225 32253 32315 32431 32432 32434 32544 33225 33254 33551 34144 34322 34452 34554 35214 35233 35453 35535 41313 41322 41424 41533 42144 42312 42324 42414 42415 42423 43111 43144 43235 44142 44225 44232 44235 44521 45212 45315 45331 45333 45344 45413 45531 51235 52121 52144 52233 52332 52342 52424 53124 53151 53221 54423 54424 54514 54524 54533 54553 55221 55242 55322 55414 55442
Case #93: NONE
Case #94: 2516717
Case #95: 11 12 21 22
Case #96: 1
Case #97: NONE
Case #98: NONE
Case #99: 216435 311115 336111 535441 631615 651311
Case #100: 111 112 113 121 122 123 131 132 133 211 212 213 221 222 223 231 232 233 311 312 313 321 322 323 331 332 333
