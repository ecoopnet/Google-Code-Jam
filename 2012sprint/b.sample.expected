Case #1: 3 7
Case #2: NONE
Case #3: 72 89
Case #4: 123 333
